plugins/tableselection/plugin.js

```diff
@@ -889,6 +857,56 @@
 		setTimeout( function() {
 			editor.fire( 'afterPaste' );
 		}, 0 );
+
+		function shouldUseCustomPaste( selection, selectedCells ) {
+			var ranges = selection.getRanges(),
+				table = ranges.length && ranges[ 0 ]._getTableElement( { table: 1 } ),
+				isIgnoredTable = table && table.hasAttribute( ignoredTableAttribute ),
+				collapsedPartial = selectedCells.length === 1 && !rangeContainsTableElement( selection.getRanges()[ 0 ] ) && !boundarySelection,
+				boundaryWithoutTable = boundarySelection && !pastedTable;
+
+			// (#2945)
+			if ( isIgnoredTable ) {
+				return false;
+			}
+
+			var selectionAncestor = selection.getCommonAncestor(),
+				selectionExceedesTable = selectionAncestor &&
+					selectionAncestor.is &&
```

Comment:
Yes, [getCommonAncestor()] can return [`CKEDITOR.dom.text`](https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_dom_text.html) (e.g. when you select text with a space between - like `So[me te]xt` ) which doesn't have `is()` method. I'll go with checking node type then. On the matter of the `getCommonAncestor()` method I think it would be better to just edit docs, but for now I'll just report issue and wait for decision.